src/xma/src/xmaapi/xmaapi.cpp

```diff
@@ -210,6 +212,8 @@ void xma_thread1() {
         }
     }
     //Print all stats here
+    std::lock_guard lock(g_xma_singleton->m_mutex);
```

Comment:
@Qizot, can we move this line to 104 line? 

if we move this to line number 104, lock_gaurd at line number 112 is not needed.

Reply:
I think that it is best to keep it that way to keep the lock congestion low. If we unlock the mutex regularly we can let other threads do something with the singleton as well.

Reply:
ok.